### DISCUSSION TITLE
Typing Improvement Tool for MonkeyType (for user)

### DISCUSSION THREAD

**Discussion by prince-patel23** [monkeytypegame/monkeytype#5031] [Ideas]
Dear Mentors,

I'm thrilled to share a new tool idea for MonkeyType that could really boost users' typing skills. It's all about helping users spot and fix mistakes as they type, and then giving them a clear picture of where they can improve afterwards.

Basically, while users are practicing, if they make a mistake, like hitting the wrong key, the program will show it in red so they can fix it right away. Then, after they're done, they'll get a graph showing which letters they tend to mess up the most. This way, they can focus on those trouble spots and get better faster.We use previous test data also to show this graph.

I think adding this feature could make MonkeyType even more helpful, and I'd love to hear what you think about it.

**"This feature is poised to significantly enhance typing speed and facilitate the development of precise muscle memory, thereby optimizing users' typing proficiency."**

![Screenshot 2024-02-09 144006](https://github.com/monkeytypegame/monkeytype/assets/151719092/e4f6fe7c-fefa-4053-a564-5631d24024c5)

Best,
@prince-patel23

**Comment by Miodec** [comment]
Its a planned feature.

**Reply by realbakke** [reply]
you guys have the best typing site its soo good
